presto-hive/src/main/java/com/facebook/presto/hive/util/Statistics.java

```diff
@@ -325,6 +328,44 @@ private static long convertLocalToUtc(DateTimeZone timeZone, long value)
         return Decimals.encodeUnscaledValue(unscaled);
     }
 
+    public static PartitionStatistics getEmptyPartitionStatistics(Map<String, Set<ColumnStatisticType>> columnStatisticsMetadatas)
+    {
+        Map<String, HiveColumnStatistics> columnStatistics = columnStatisticsMetadatas.entrySet().stream()
+                .collect(toImmutableMap(Entry::getKey, entry -> createColumnStatisticsForEmptyPartition(entry.getValue())));
+        return new PartitionStatistics(createZeroStatistics(), columnStatistics);
+    }
+
+    private static HiveColumnStatistics createColumnStatisticsForEmptyPartition(Set<ColumnStatisticType> columnStatisticTypes)
+    {
+        HiveColumnStatistics.Builder result = HiveColumnStatistics.builder();
+        for (ColumnStatisticType columnStatisticType : columnStatisticTypes) {
+            switch (columnStatisticType) {
+                case MAX_VALUE_SIZE_IN_BYTES:
+                    result.setMaxValueSizeInBytes(0);
+                    break;
+                case TOTAL_SIZE_IN_BYTES:
+                    result.setTotalSizeInBytes(0);
+                    break;
+                case NUMBER_OF_DISTINCT_VALUES:
+                    result.setDistinctValuesCount(0);
+                    break;
+                case NUMBER_OF_NON_NULL_VALUES:
+                    result.setNullsCount(0);
+                    break;
+                case NUMBER_OF_TRUE_VALUES:
+                    result.setBooleanStatistics(new BooleanStatistics(OptionalLong.of(0L), OptionalLong.of(0L)));
+                    break;
+                case MIN_VALUE:
+                case MAX_VALUE:
+                    // leave min/max value as default which is Optional.empty()
```

Comment:
Currently if column is empty (either has no values, or all values are `null`) , the `setMinMax` still set's the `IntegerStatistics` and similar objects to `new IntegerStatistics(Optional.empty(), Optional.empty())`. Please create a `setMinMaxForEmptyPartition(Type columnType, HiveColumnStatistics.Builder result)` that does the same, to be consistent.

Reply:
Agree. `IntegerStatistics` is `Optional.empty()` means that the column is not of `Integer` type. `IntegerStatistics(Optional.empty(), Optional.empty())` means the column is of `Integer` type and the stat is empty.